src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -8,33 +11,64 @@ import {
 } from 'platform/user/authentication/selectors';
 import { autoLogin, autoLogout } from 'platform/user/authentication/utilities';
 import { hasSession, hasSessionSSO } from 'platform/user/profile/utilities';
-import { ssoKeepAliveSession } from 'platform/utilities/sso';
+import {
+  ssoKeepAliveSession,
+  getForceAuth,
+  setForceAuth,
+  deleteForceAuth,
+} from 'platform/utilities/sso';
+
+function parseqs(value) {
+  /*
+   * naive query string parsing function, takes a query string and returns
+   * an object mapping the keys to values.
+   */
+  const data = value.startsWith('?') ? value.substring(1) : value;
+  const entries = data ? data.split('&').map(q => q.split('=')) : [];
+  return fromPairs(entries);
+}
 
 export async function checkStatus(toggleKeepAlive) {
   await ssoKeepAliveSession();
   if (hasSession() && !hasSessionSSO()) {
     autoLogout();
-  } else if (!hasSession() && hasSessionSSO()) {
+  } else if (!hasSession() && hasSessionSSO() && !getForceAuth()) {
     autoLogin();
   }
 
   toggleKeepAlive();
 }
 
-function AutoSSO(props) {
-  const { useSSOe, useInboundSSOe, hasCalledKeepAlive } = props;
+class AutoSSO extends React.Component {
+  render() {
+    const {
+      useSSOe,
+      useInboundSSOe,
+      hasCalledKeepAlive,
+      userLoggedIn,
+    } = this.props;
 
-  if (useSSOe && useInboundSSOe && !hasCalledKeepAlive) {
-    checkStatus(props.checkKeepAlive);
-  }
+    if (parseqs(window.location.search).auth === 'force-needed') {
+      setForceAuth();
```

Comment:
this makes me think about some of the inbound issues, too...if the purpose of this is to make sure we detect query params sent back to `/auth/login/callback`, this might hit a race condition with `AuthApp`'s redirect call and it might redirect to another page before this can get executed

Reply:
@hinzed1127 do you think I should update the logic to be ....
```
const auth = parseqs(window.location.search).auth
if (auth && auth !=== 'success') {
  setForceAuth();
}
```